code/modules/mob/living/simple_animal/hostile/hostile.dm

```diff
@@ -42,16 +42,8 @@
 			break
 
 		if(isliving(A))
-			var/mob/living/L = A
-			if((L.faction == src.faction) && !attack_same)
-				continue
-			else if(L in friends)
-				continue
-			else
-				if(!L.stat)
-					stance = HOSTILE_STANCE_ATTACK
-					T = L
-					break
+			stance = HOSTILE_STANCE_ATTACK
+			T = find_weakest()

```

Comment:
So, you're calling `ListTargets()` once for every mob in range? This will be comically expensive.